qemu/tests/pcie_hotplug_opt.py

```diff
@@ -0,0 +1,132 @@
+import logging
+
+from virttest import error_context
+from virttest.qemu_monitor import QMPCmdError
+
+
+@error_context.context_aware
+def run(test, params, env):
+    """
+    Hot-unplug/Hot-plug virtio-blk-pci, virtio-scsi-pci, virtio-net-pci while
+    the parent pcie-root-port use 'hotplug=off'
+
+    1) Boot guest with virtio-blk-pci, virtio-scsi-pci, virtio-net-pci device,
+       and the parent pcie-root-port use 'hotplug=off'
+    2) Hot unplug those devices
+    3) Hot plug new virtio-blk-pci, virtio-scsi-pci, virtio-net-pci device to
+       pcie-root-port that use 'hotplug=off'
+
+    :param test:   QEMU test object.
+    :param params: Dictionary with the test parameters.
+    :param env:    Dictionary with test environment.
+    """
+
+    def hotplug_blk():
+        """
+        Hot-plug virtio-blk-pci
+        """
+        virtio_blk_pci_dev = image_devs[-1]
+        virtio_blk_pci_dev.set_param("bus", free_root_port_id)
+        virtio_blk_pci_dev.hotplug(vm.monitor)
+
+    def hotplug_scsi():
+        """
+        Hot-plug virtio-scsi-pci
+        """
+        pci_add_cmd = "device_add driver=virtio-scsi-pci, id=plug"
+        pci_add_cmd += ",bus=%s" % free_root_port_id
+        vm.monitor.send_args_cmd(pci_add_cmd)
```

Comment:
Would you tell why not use `simple_hotplug()` or `device.hotplug()`?

Reply:
For virtio-scsi-pci, hotplug the virtio-scsi-pci controller can cover the test scenario, so i did not add params to prepare a scsi disk for hotplug, it is redundant i think.  So no device object was created. 

I found `simple_hotplug` does not release the free root_port even if hotplug failed, so i did not use it for hotplug virtio-blk disk. Otherwise, hotplug nic need add a new pci-root-port.

Reply:
@yanan-fu I see, it's acceptable for me. Thanks for the explanation.